File: src/main/java/org/pixeltime/enchantmentsenhance/listener/VanillaEnchantHandler.java
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */
package org.pixeltime.enchantmentsenhance.listener;


import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;
import org.pixeltime.enchantmentsenhance.gui.menu.Menu;


public class VanillaEnchantHandler implements Listener {
    @EventHandler(priority = EventPriority.HIGHEST)
    public void onOpenEnchantmentTable(PlayerInteractEvent event) {
        if (event.getAction() == Action.RIGHT_CLICK_BLOCK && event.getClickedBlock().getType() == Material.ENCHANTMENT_TABLE) {
            event.setCancelled(true);
            new Menu(event.getPlayer()).open();
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/listener/VanillaEnchantHandler.kt
package org.pixeltime.enchantmentsenhance.listener

import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.player.PlayerInteractEvent
import org.pixeltime.enchantmentsenhance.event.menu.Menu
import org.pixeltime.enchantmentsenhance.manager.SettingsManager


class VanillaEnchantHandler : Listener {
    private val disableVanilla = SettingsManager.config.getBoolean("disableVanillaEnchant")

    @EventHandler(priority = EventPriority.HIGHEST)
    fun onOpenEnchantmentTable(event: PlayerInteractEvent) {
        if (disableVanilla && event.action == Action.RIGHT_CLICK_BLOCK && event.clickedBlock.type == Material.ENCHANTMENT_TABLE) {
            event.isCancelled = true
            Menu.showEnhancingMenu(event.player)
        }
    }
}